src/ts/state/test/reducer.spec.ts

```diff
@@ -22,37 +22,63 @@ import {
 import { expandRangesToParentBlockNode } from "../../utils/range";
 import { createDoc, p } from "../../test/helpers/prosemirror";
 import { IMatch, IMatchRequestError } from "../../interfaces/IMatch";
+import { addMatchesToState } from "../helpers";
 import {
   createMatcherResponse,
   createBlock,
   exampleCategoryIds,
   createBlockQueriesInFlight,
   exampleRequestId,
-  createInitialData,
+  createInitialState,
   defaultDoc,
   createMatch,
-  addMatchesToState
+  ICreateMatcherResponseSpec
 } from "../../test/helpers/fixtures";
 import { createBlockId } from "../../utils/block";
+import { getBlocksFromDocument } from "../../utils/prosemirror";
 
-const reducer = createReducer(expandRangesToParentBlockNode);
+const reducer = createReducer(expandRangesToParentBlockNode, () => true);
+
+/**
+ * Create a plugin state, creating the given matches and
+ * their decorations from the given spec.
+ */
+const createStateWithMatches = (
```

Comment:
There's a bit of scaffolding necessary to test the ignoreMatches predicate, which I've abstracted here. This is likely to be useful for other tests, but I haven't included that change here to keep the diff manageable.